pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -710,6 +712,49 @@ outLoop:
 	return nil
 }
 
+func (h *appHandler) handleRollout(ctx context.Context) (reconcile.Result, error) {
+	var comps []string
+	for _, component := range h.app.Spec.Components {
+		comps = append(comps, component.Name)
+	}
+
+	// targetRevision should always points to LatestRevison
+	targetRevision := h.app.Status.LatestRevision.Name
+	var srcRevision string
+	target, _ := oamutil.ExtractRevisionNum(targetRevision)
+	// if target == 1 this is a initial scale operation, sourceRevision should be empty
+	// otherwise source revision always is targetRevision - 1
```

Comment:
please make sure this case have e2e test.

Reply:
Got it